Source/FieldSolver/FiniteDifferenceSolver/HybridPICModel/HybridPICModel.cpp

```diff
@@ -125,6 +128,30 @@ void HybridPICModel::AllocateLevelMFs (
         lev, amrex::convert(ba, jz_nodal_flag),
         dm, ncomps, ngJ, 0.0_rt);
 
+    // Allocate species specific multifabs to deposit individual currents
+    for (const auto & specname : m_ion_species_names)
```

Comment:
It would be good to add a check in here for species charge to not allocate all the extra space for neutral particles (and then similarly skip the deposition, synchronization, guard cell exchange and addition to J_total in `DepositRhoAndJ`).